File: 1-Beginner/1061/bee1061.c
// 1061 - Tempo de um Evento
#include <stdio.h>
#include <string.h>

int main(void) {

    int d1, h1, m1, s1, d2, h2, m2, s2;
    char lixo[5];

    char input[15]; 
    fgets(input, 15, stdin);
    input[strcspn(input, "\n")] = '\0';
    sscanf(input, "%s %d", lixo, &d1);

    fgets(input, 15, stdin);
    input[strcspn(input, "\n")] = '\0';
    sscanf(input, "%d %c %d %c %d", &h1, lixo, &m1, lixo, &s1);

    fgets(input, 15, stdin);
    input[strcspn(input, "\n")] = '\0';
    sscanf(input, "%s %d", lixo, &d2);

    fgets(input, 15, stdin);
    input[strcspn(input, "\n")] = '\0';
    sscanf(input, "%d %c %d %c %d", &h2, lixo, &m2, lixo, &s2);

    int inicioH = h1*3600 + m1*60 + s1;
    int fimH = h2*3600 + m2*60 + s2; 

    int dia, hora, min, seg;

    if(inicioH < fimH) {
        dia = d2 - d1;
        seg = fimH - inicioH;
        hora = seg/3600;   seg -= hora*3600;
        min = seg/60;      seg -= min*60;
    }
    else if(inicioH > fimH) {
        dia = d2 - d1 - 1;
        seg = 86400 - inicioH + fimH;
        hora = seg/3600;   seg -= hora*3600; 
        min = seg/60;      seg -= min*60;
    }
    else {
        dia = d2 - d1;
        hora = 0; min = 0; seg = 0;
    }

    printf("%d dia(s)\n%d hora(s)\n%d minuto(s)\n%d segundo(s)\n", dia, hora, min, seg);

    return 0;
}

File: 1-Beginner/1061/bee1061.java
// 1061 - Tempo de um Evento
import java.io.IOException;
import java.util.Scanner; 

public class bee1061 {
 
    public static void main(String[] args) throws IOException {
        Scanner scan = new Scanner(System.in);
        
        String[] dia1 = scan.nextLine().split(" ");
        String[] hora1 = scan.nextLine().split(" ");
        String[] dia2 = scan.nextLine().split(" ");
        String[] hora2 = scan.nextLine().split(" ");
        
        int inicioH = Integer.parseInt(hora1[0])*3600 + Integer.parseInt(hora1[2])*60 + Integer.parseInt(hora1[4]);
        int fimH = Integer.parseInt(hora2[0])*3600 + Integer.parseInt(hora2[2])*60 + Integer.parseInt(hora2[4]);
    
        int inicioD = Integer.parseInt(dia1[1]);
        int fimD = Integer.parseInt(dia2[1]);
        
        int d, h, m, s;
        if(inicioH < fimH) {
            d = fimD - inicioD;
            s = fimH - inicioH;
            h = s/3600;   s -= h*3600;
            m = s/60;     s -= m*60;
        }
        else if(inicioH > fimH) {
            d = fimD - inicioD - 1;
            s = 86400 - inicioH + fimH;
            h = s/3600;   s -= h*3600; 
            m = s/60;     s -= m*60;
        }
        else {
            d = fimD - inicioD;
            h = 0; m = 0; s = 0;
        }

        System.out.println(d + " dia(s)\n" + h + " hora(s)\n" + m + " minuto(s)\n" + s + " segundo(s)");
    
        scan.close();
    }
 
}